src/Interface.jl

```diff
@@ -1,38 +1,36 @@
-
-
-# Functions
-occurrences(com::AbstractAssemblage) = error("function not defined for this type") #has to be implemented with the concrete type
-thingnames(com::AbstractAssemblage) = error("function not defined for this type")
+# Functions - most have to be implemented with the concrete type
+occurrences(com::AbstractAssemblage) = error("function not defined for this type") 
 view(com::AbstractAssemblage) = error("function not defined for this type")
 places(asm::AbstractAssemblage) = error("function not defined for this type")
 things(asm::AbstractAssemblage) = error("function not defined for this type")
-nplaces(asm::AbstractPlaces) = error("function not defined for this type")
+
+nplaces(com::AbstractPlaces) = size(occurrences(com), 2)
 placenames(com::AbstractPlaces) = error("function not defined for this type")
 
-nzrows(a::AbstractMatrix) = find(sum(a .> 0, 2)> 0)
-nzcols(a::AbstractMatrix) = find(sum(a .> 0, 1)> 0)
+nthings(com::AbstractThings) = size(occurrences(com), 1)
+thingnames(com::AbstractThings) = error("function not defined for this type")
+
+nzrows(a::AbstractMatrix) = find(sum(a .> 0, 2) > 0)
+nzcols(a::AbstractMatrix) = find(sum(a .> 0, 1) > 0)
 
 occurring(com::AbstractAssemblage) = nzrows(occurrences(com))
 occupied(com::AbstractAssemblage) = nzcols(occurrences(com))
-occupied(com::AbstractAssemblage, idx) = findn(occurrences(com)[idx,:])
-occurring(com::AbstractAssemblage, idx) = findn(occurrences(com)[:,idx])
+occupied(com::AbstractAssemblage, idx) = findn(occurrences(com)[idx, :])
+occurring(com::AbstractAssemblage, idx) = findn(occurrences(com)[:, idx])
 
 noccurring(x) = length(occurring(x))
 noccupied(x) = length(occupied(x))
 noccurring(x, idx) = length(occurring(x, idx))
 noccupied(x, idx) = length(occupied(x, idx))
 
-nthings(com::AbstractThings) = size(occurrences(com), 1)
-nplaces(com::AbstractPlaces) = size(occurrences(com), 2)
-
 thingoccurrences(com::AbstractAssemblage, idx) = view(occurrences(com), idx, :)
 placeoccurrences(com::AbstractAssemblage, idx) = view(occurrences(com), :, idx) # make certain that the view implementation also takes thing or place names
 
-richness(com::AbstractAssemblage{Bool}) = vec(sum(occurrences(com), 1))
-richness(com::AbstractAssemblage{<:Real}) = vec(mapslices(nnz, occurrences(com), 1))
+richness(com::AbstractAssemblage{_, _, Bool}) = vec(sum(occurrences(com), 1))
```

Comment:
I think the head honchos are actively discouraging using `_` as placeholders like this, because it may get a syntax function in the future (e.g. for implementing short anonymous functions)